spk/rutorrent/Makefile

```diff
@@ -27,35 +53,64 @@ SERVICE_USER = auto
 SERVICE_SETUP = src/service-setup.sh
 SERVICE_WIZARD_SHARE = wizard_download_dir
 
-SERVICE_COMMAND = /var/packages/rutorrent/target/bin/rtorrent-daemon
+SERVICE_COMMAND = screen -D -m rtorrent
 FWPORTS = src/rutorrent.sc
 CONF_DIR = src/conf/
-WIZARDS_DIR = src/wizard/
+WIZARDS_DIR = $(WORK_DIR)/generated-wizards
+WIZARDS = install_uifile upgrade_uifile
 
 INSTALL_DEP_SERVICES = apache-web
 START_DEP_SERVICES = apache-web
 INSTUNINST_RESTART_SERVICES = apache-web
 
 POST_STRIP_TARGET = rutorrent_extra_install
 
-BUSYBOX_CONFIG = usrmng nice procutils
 ENV += BUSYBOX_CONFIG="$(BUSYBOX_CONFIG)"
 
 # some dependents require C++11 support
 UNSUPPORTED_ARCHS = $(ARMv5_ARCHS) $(OLD_PPC_ARCHS)
 # unsupported with cross/rtorrent
 UNSUPPORTED_ARCHS += $(ARMv7L_ARCHS)
 
+SYSTEM_GROUP = http
+
+ICON_DIR = $(STAGING_DIR)/$(DSM_UI_DIR)/images
```

Comment:
This is a hack to force the generation of icons. A rework of the framework could be useful to force them (otherwise DSM 7 specific resource workers wouldn't be using available icons)

Reply:
I'm not sure I follow, what's missing exactly?

Reply:
What's missing is that without this trick the icons are not automatically generated. Will rehave a look because framework has evolved lately but it seems icons were only generated in some very specific context.

Reply:
looking at spksrc.icon.mk, icons are created when DSM_UI_DIR and SPK_ICON and SERVICE_PORT are defined and NO_SERVICE_SHORTCUT is not defined.

I suppose this will be fixed by adding to Makefile:
`SERVICE_PORT = 8050`

Reply:
Won't this be causing issues with the interactions with the way you can "open" the page ? (Checking)

Reply:
And probably you can use the SERVICE_PORT variable when patching the scgi_port in config.php at installation time.

Reply:
> Won't this be causing issues with the interactions with the way you can "open" the page ? (Checking)

I suppose it does not. It could change the generation of the app/config file, but you have a custom app/config file that is in src/app/config and is installed in rutorrent_extra_install. So you have to check that this file is still the same when adding SERVICE_PORT definition.

Reply:
@smaarn, is this concern regarding the package icon generation now resolved?

Reply:
@mreid-tt issue should now be fixed. Thanks for following up on this and sorry for not answering earlier.

@hgy59  Found a much more more elegant way by the way to keep my app/config customization (by using an already existing framework option